### DISCUSSION TITLE
Is activej fully "self-sufficient" framework - security, validation, database,...?

### DISCUSSION THREAD

**Discussion by 0248991** [activej/activej#201] [Q&A] [answered]
Hey,
On the main page, ActiveJ is described as "an alternative to Spring/Micronauts/Netty/Jetty" and "a self-sufficient (no third-party dependencies) platform".

The documentation mentions only performance features - like IO, threads, HTTP, serializer, bytecode manipulation,... - but I don't see anything about the capabilities of a full-fledged framework - like security(auth, session, rights,...), database access (what about ORM and entity (de)serialization using these super fast tools?), (POJO) validation, exception handlers, etc...?

I am confused because I haven't found anything in the docs as well as in online articles; and no actual beginner-friendly tutorials like a basic Hello World end-to-end example.

Can you clarify these points and what is the state of ActiveJ?

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @0248991
ActiveJ is a technology agnostic framework. Essentially, it is a full-fledged alternative network I/O stack, including high-performance HTTP client and server, asynchronous IO, promises, Eventloop (not just a web framework built on top of some preexisting HTTP server). What it means is that we do not dictate what library you can use for security, ORM, JSON processing, etc. You can use third-party libraries of your choice on top of ActiveJ components to add the functionality that your business logic requires.
